index.js

```diff
@@ -1,85 +1,69 @@
 require('dotenv').config();
 
-var accessToken = process.env.YOUR_INSTAGRAM_ACCESS_TOKEN;
-const user_id = process.env.USER_ID_TO_MONITOR;
-var slack_url = process.env.SLACK_INCOMING_WEBHOOK_URL;
+const accessToken = process.env.YOUR_INSTAGRAM_ACCESS_TOKEN;
+const userID = process.env.USER_ID_TO_MONITOR;
+const slackURL = process.env.SLACK_INCOMING_WEBHOOK_URL;
 
-var InstagramAPI = require('instagram-api');
-var request = require('request');
-var instagramAPI = new InstagramAPI(accessToken);
-var request = require('request');
-const express = require('express')
-const app = express()
+const InstagramAPI = require('instagram-api');
+const request = require('request');
 
+const instagramAPI = new InstagramAPI(accessToken);
+const express = require('express');
 
-app.get('/run', function (req, res)
-{
-    instagramAPI.userMedia(user_id).then(result =>
+const app = express();
+
+function sendPostToSlack(media) {
+  let caption;
+  if (media.caption === null) {
+    caption = 'Unknown';
+  } else {
+    caption = media.caption.text;
+  }
+  request.post(
+    slackURL,
     {
-        var pics = [];
-        var likePromise = [];
-        result.data.forEach(function(media){
+      json: {
+        attachments: [
+          {
+            pretext: caption,
+            image_url: media.images.standard_resolution.url,
+            footer: `Instagram ${media.type} liked`,
+            footer_icon: 'https://lh3.googleusercontent.com/aYbdIM1abwyVSUZLDKoE0CDZGRhlkpsaPOg9tNnBktUQYsXflwknnOn2Ge1Yr7rImGk=w300',
+            ts: Math.floor(Date.now() / 1000),
+          },
+        ],
+      },
+    },
+    (error, response) => response,
+  );
+}
 
-        if (media.user_has_liked == false)
-            likePromise.push(instagramAPI.postMediaLike(media.id).then(resp =>
-            {
-                return sendPostToSlack(media);
-            }));
-        });
 
-        Promise.all(likePromise).then(values =>
-            {
-                console.log('All photos liked')
-                res.send(values);
-            },
-            function(err)
-            {
-                res.send(err);
-            }
-        );
+app.get('/run', (req, res) => {
+  instagramAPI.userMedia(userID).then(
+    (result) => {
+      const likePromise = [];
+      result.data.forEach((media) => {
+        if (media.user_has_liked === false) {
+          likePromise.push(instagramAPI.postMediaLike(media.id).then(() =>
```

Comment:
Maybe something like:

```javascript
likePromise.push(
  instagramAPI
    .postMediaLike(media.id)
    .then(() => sendPostToSlack(media)),
);
```

Little bit easier on the eyes 😄

Reply:
userHasLiked - whats this?

Reply:
It's declared here: ` user_has_liked: userHasLiked,`

Pattern matching is available when destructuring 😄

Reply:
Nice.. new one into my knowledge book.

Reply:
Yep, very useful when dealing with non JS backend APIs